core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -167,5 +169,18 @@ private static String pathContext(String tableLocation) {
 
       return resolvedContext;
     }
+
+    private static String computeHash(String fileName) {
```

Comment:
This is okay, but I generally prefer using a library, like the URL-safe base64 variant. To avoid `=`, I think we can make the encoded value a multiple of 3 bytes long. Then we just add - and _.

Reply:
@singhpk234 It looks like you resolved this, but I think we still need to address the `=` case.  I assume we can just pad the bytes to make sure we don't have URL padding.

Reply:
ACK made the changes, by using the HASH buffer of 6 bytes suggested below.